Dockerfile

```diff
@@ -0,0 +1,26 @@
+ARG JENKINS_SLAVE
+FROM jenkins/jnlp-slave:${JENKINS_SLAVE}
+USER root
+
+ARG PYTHON_MAJOR
+ARG PYTHON_PATCH
+RUN export PYTHON_SEMVER="${PYTHON_MAJOR}.${PYTHON_PATCH}" && \
+  apt-get update && \
+  apt-get install wget apt-utils build-essential -y && \
+  apt-get install libreadline-dev libncursesw5-dev libssl-dev \
+  libsqlite3-dev tk-dev libgdbm-dev libc6-dev libbz2-dev libffi-dev zlib1g-dev -y && \
+  cd /usr/src && \
+  wget https://www.python.org/ftp/python/${PYTHON_SEMVER}/Python-${PYTHON_SEMVER}.tgz && \
+  tar xzf Python-${PYTHON_SEMVER}.tgz && \
+  rm Python-${PYTHON_SEMVER}.tgz && \
+  cd Python-${PYTHON_SEMVER} && \
+  ./configure --enable-optimizations --with-ensurepip && \
+  make altinstall && \
+  pip${PYTHON_MAJOR} install six && \
+  # Clean up
+	apt-get remove -f -y --purge --auto-remove build-essential && \
```

Comment:
As a side note regarding `build-essentials`:

Some packets that are installed via `pip` do require gcc and other build tools as well as Python `*-dev` packages in order to compile... just to keep this in mind.

Reply:
We will test it